.github/workflows/test.yaml

```diff
@@ -350,6 +385,28 @@ jobs:
             echo "No spread log found, skipping errors reporting"
         fi
 
+    - name: save spread results
+      if: always()
+      run: |
+          if [ -f spread.log ]; then
+              echo "Running spread log analyzer"
+
+              # Aborted cannot be used to determine if we need to re-run all because it could be a aborted an entire suite
+              if grep -q "Aborted tasks: 0" spread.log; then
+                  ./tests/lib/external/snapd-testing-tools/utils/log-parser spread.log --output spread-results.json
+                  FAILED_TESTS="$(jq -r '.[] | select( .type == "result") | select( .result_type == "Failed") | select( .level == "task" or .level == "tasks" ) | .detail.lines[]' spread-results.json | cut -d '-' -f2- | xargs)"
+
+                  echo "Failed tests saved for re-execution: $FAILED_TESTS"
+                  echo "$FAILED_TESTS" > "$FAILED_TESTS_FILE"
```

Comment:
Are we sure that we won't hit a limit on command line length (or number of parameters) when calling `xargs`? I've seen that happening sometimes when running `find | xargs ...`, but I don't know what the limit is.
Can't we just do
```
jq -r '.[] | select( .type == "result") | select( .result_type == "Failed") | select( .level == "task" or .level == "tasks" ) | .detail.lines[]' spread-results.json \
    | cut -d '-' -f2- > "$FAILED_TEST_FILE"
```
?

Reply:
agree, updated, thanks for the review